modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -1113,6 +1115,94 @@ void TFImporter::parseReshape(tensorflow::GraphDef& net, const tensorflow::NodeD
     }
 }
 
+void TFImporter::parseExpandDims(tensorflow::GraphDef& net, const tensorflow::NodeDef& layer, LayerParams& layerParams)
+{
+      const std::string& name = layer.name();
+    const int num_inputs = layer.input_size();
+
+    CV_Assert(!netInputShapes.empty());
```

Comment:
When `netInputShapes` is empty, it will not work.

Reply:
I'm not sure there is. I tried adding shape propagation to TF importer some time ago, similar to ONNX importer, but realized that netInputShapes aren't required to exist.